python/hyperspace/hyperspace.py

```diff
@@ -0,0 +1,173 @@
+import sys
+
+from pyspark.sql import SparkSession, DataFrame
+from hyperspace import *
+from py4j.java_gateway import java_import
+
+
+class Hyperspace:
+    def __init__(self, spark):
+        """
+        Initializes hyperspace object.
+        :param spark: sparkSession object
+        :return: hyperspace object
+
+        >>> hyperspace = Hyperspace(spark)
+        """
+        self.sparkContext = spark.sparkContext
+        self.spark = spark
+        self.jvm = spark._jvm
+        self.hyperspace = self.jvm.com.microsoft.hyperspace.Hyperspace(spark._jsparkSession)
+
+    def __getJavaIndexConfig(self, index_config):
```

Comment:
Are you deliberately putting these private helper methods before the public API methods?  PEP-8 doesn't have guidance about this so I'm wondering what you're following.

Reply:
I was defining the private functions before their use in public functions. One reason to avoid naming error mentioned here I initially got in testing.  
https://stackoverflow.com/questions/45025005/python-function-calling-order

Also I saw in pyspark code base also does not follow particular order to differentiate public/private methods. Just functions are defined before use. Eg

https://github.com/apache/spark/blob/master/python/pyspark/sql/session.py